### PR TITLE
Make protections testable

### PR DESCRIPTION
**Pull request by kdzwinel**
This PR modifies some of the content scope scripts to allow passing custom `window` object. This is to support testing - I want to control the whole "browser" environment so scripts have to use `globalThis`, which can be overwritten, instead of `window`.

https://app.asana.com/0/0/1201702136255075/f

### CODE REVIEW

File: src/cookie.js [hunk 1]
```diff
@@ -51,5 +51,5 @@ export class Cookie {
 
 // support node-requires (for test import)
 if (typeof module !== 'undefined' && module.exports) {
-    module.exports = Cookie
```
Comment (kdzwinel) [inline src/cookie.js:54]: this is to match how this class is exported at the top:

> export class Cookie

makes sure that export is the same in browser vs node

Comment (jonathanKingston) [reply src/cookie.js:54]: Can we just make the test es6 modules compliant so we can remove this instead?

**Comment by shakyShane** [comment]
@kdzwinel 👋🏻 

I very much appreciate the effort to increase testing here :)

I'd like to suggest that we don't alter any method signatures though, at least not for the purpose of testing - for the following 2 reasons.

1 - the content-script features have a very small API surface currently, and I believe we should strive to keep that as small as possible. Choosing to take up a second-position parameter, adding `globalThis`, limits our choices for API changes in the future. An object would be better here, but honestly I'd rather that not contain environmental things.

2 - if we create a codebase where browser globals are accessed locally in individual function signatures, it creates a situation where engineers on new part of the codebase may not be sure whether or not to access things globally, or to plumb them in as you had to do in `src/features/fingerprinting-screen-size.js` for example.

**Proposal:**

Although I'm not usually a fan of global mocks, I believe the scope of this project warrants a singular way for accessing/testing `globalThis` - without changing any defined APIs.

Could we just install JSDOM's globals on an as-needed basis, like this:

https://github.com/duckduckgo/content-scope-scripts/commit/4d5d1952220ee8e24393146d5d72ef6adf889989

As @jonathanKingston has mentioned in the Asana task, if we were to pass `globalThis` around, there would be other places where we'd need to consider global access too, which this approach would solve too.

@kdzwinel what do you think?

**Comment by shakyShane** [comment]
Also worth noting that I took some time to ensure features can be tested within an extension - even with the ability to provide config within the test, please see https://github.com/duckduckgo/content-scope-scripts/blob/main/integration-test/test-navigator-interface.js#L24

File: src/cookie.js (hunk 1 above)
Comment (kdzwinel) [reply src/cookie.js:54]: Yeah, good point. I ended up removing and… everything still seems to work? Both here and in extension tests pass.

File: src/features/tracking-cookies-1p.js [hunk 2]
```diff
@@ -116,6 +118,8 @@ function applyCookieExpiryPolicy () {
 
 // Set up 1st party cookie blocker
 export function load (args) {
+    trackerHosts.clear()
```
Comment (kdzwinel) [inline src/features/tracking-cookies-1p.js:121]: this is not related to globals, but related to testing. This file has trackerHosts variable which was retaining state between tests, there was no way to clear it.

**Comment by kdzwinel** [comment]
Thanks both for comments, I updated the PR as discussed. PTAL.

**Review by sammacbeth — DISMISSED** [review]
This looks good to me. I think you should commit the updated built files though too.

**Review by sammacbeth — APPROVED** [review]
👍

**Pull request merged**